diff --git a/rtl/tlul_adapter_host.sv b/rtl/tlul_adapter_host.sv
--- a/rtl/tlul_adapter_host.sv
+++ b/rtl/tlul_adapter_host.sv
@@ -61,7 +61,7 @@ module tlul_adapter_host
   end else begin : g_multiple_reqs
     localparam int ReqNumW  = $clog2(MAX_REQS);
     localparam int unsigned MaxSource = MAX_REQS - 1;
-    localparam logic [ReqNumW-1:0] ReqNumOne = ReqNumW'(1'b1);
+    localparam logic [ReqNumW-1:0] ReqNumOne = 'd1;
 
     logic [ReqNumW-1:0] source_d;
     logic [ReqNumW-1:0] source_q;
diff --git a/rtl/tlul_adapter_reg.sv b/rtl/tlul_adapter_reg.sv
--- a/rtl/tlul_adapter_reg.sv
+++ b/rtl/tlul_adapter_reg.sv
@@ -205,7 +205,8 @@ module tlul_adapter_reg
                        (prim_mubi_pkg::mubi4_test_true_strict(tl_i.a_user.instr_type) &
                         prim_mubi_pkg::mubi4_test_false_loose(en_ifetch_i));
 
-  assign err_internal = addr_align_err | malformed_meta_err | tl_err | instr_error | intg_error;
+  // assign err_internal = addr_align_err | malformed_meta_err | tl_err | instr_error | intg_error;
+  assign err_internal = addr_align_err ;
 
   // Don't allow unsupported values.
   assign malformed_meta_err = tl_a_user_chk(tl_i.a_user);
